cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -593,6 +604,250 @@ private Future<List<Condition>> maybeRestartConnectorTask(Reconciliation reconci
+     * @param host              Kafka Connect host
+     * @param apiClient         Kafka Connect REST API client
+     * @param connectorName     Name of the connector
+     * @param resource          The resource that defines the connector
+     * @param connectorSpec     Spec of the connector
+     * @param conditions        Status conditions of the resource
+     *
+     * @return Future with conditions which completes when the connector offset management is complete
+     */
+    @SuppressWarnings({ "rawtypes" })
+    /* test */ Future<List<Condition>> manageConnectorOffsets(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, CustomResource resource, KafkaConnectorSpec connectorSpec, List<Condition> conditions) {
+        KafkaConnectorOffsetsAnnotation annotation;
+        try {
+            annotation = getConnectorOffsetsOperation(resource, connectorName);
+        } catch (InvalidResourceException e) {
+            String message = "Encountered error getting connector offsets annotation. " + e.getMessage();
+            LOGGER.warnCr(reconciliation, message);
+            conditions.add(StatusUtils.buildWarningCondition("ManageOffsets", message));
+            return Future.succeededFuture(conditions);
+        }
+
+        switch (annotation) {
+            case list -> {
+                return listConnectorOffsets(reconciliation, host, apiClient, connectorName, resource, connectorSpec, conditions);
+            }
+            case alter -> {
+                return alterConnectorOffsets(reconciliation, host, apiClient, connectorName, resource, connectorSpec, conditions);
+            }
+            case reset -> {
+                return resetConnectorOffsets(reconciliation, host, apiClient, connectorName, resource, conditions);
+            }
+            default -> {
+                return Future.succeededFuture(conditions);
+            }
+        }
+    }
+
+    /**
+     * Fetches the connector offsets and writes them to a ConfigMap.
+     * This operation requires the listOffsets property to be set on
+     * the Custom Resource giving the name of the ConfigMap to use.
+     *
+     * @param reconciliation    Reconciliation marker
+     * @param host              Kafka Connect host
+     * @param apiClient         Kafka Connect REST API client
+     * @param connectorName     Name of the connector
+     * @param resource          The resource that defines the connector
+     * @param connectorSpec     Spec of the connector
+     * @param conditions        Status conditions of the resource
+     *
+     * @return Future with conditions which completes when the connector offset list action is complete
+     */
+    @SuppressWarnings({ "rawtypes" })
+    Future<List<Condition>> listConnectorOffsets(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, CustomResource resource, KafkaConnectorSpec connectorSpec, List<Condition> conditions) {
```

Comment:
Is there some reason why this is not private? It does not seem to be used anywhere else.

Reply:
That was a mistake, it should be private like the other similar methods, I've updated it